src/cron-job.ts

```diff
@@ -0,0 +1,251 @@
+   * Specifies the number of successful jobs history retained.
+   *
+   * @default - 3.
+   */
+  readonly successfulJobsRetained?: number;
+
+  /**
+   * Specifies the number of failed jobs history retained.
+   *
+   * @default - 1.
+   */
+  readonly failedJobsRetained?: number;
+}
+
+/**
+ * A CronJob is responsible for creating a Job and scheduling it based on provided scheduled. This helps running Jobs in a recurring manner.
+ */
+export class CronJob extends workload.Workload {
+  /**
+   * Properties of the recurring `Job`.
+   */
+  public readonly jobProperties: JobProps;
+
+  /**
+   * Schedule for which to run the recurring job. By default, the job would be scheduled to run every minute.
+   */
+  public readonly schedule?: CronOptions;
+
+  /**
+   * Timezone in which the recurring job is scheduled.
+   */
+  public readonly timeZone?: string;
+
+  /**
+   * Concurrency policy for the recurring job. This specifies how the job scheduling would be handled based on which policy is selected.
+   */
+  public readonly concurrencyPolicy?: string;
+
+  /**
+   * Deadline to start next instance of job after the expected scheduling time. The job is considered as failed if it misses this deadline.
+   */
+  public readonly startingDeadline?: Duration;
+
+  /**
+   * Flag that tells to suspend the upcoming executions of the job.
+   */
+  public readonly suspend?: boolean;
+
+  /**
+   * The number of successful jobs to retain.
+   */
+  public readonly successfulJobsRetained?: number;
+
+  /**
+   * The number of failed jobs to retain.
+   */
+  public readonly failedJobsRetained?: number;
+
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  /**
+   * Represents the resource type.
+   */
+  public readonly resourceType = 'cronjob';
+
+  constructor(scope: Construct, id: string, props: CronJobProps) {
+    super(scope, id, {
+      restartPolicy: RestartPolicy.ON_FAILURE,
+      select: false, //Why is this false in Job?
+      ...props,
+    });
+
+    this.apiObject = new k8s.KubeCronJob(this, 'Resource', {
+      metadata: props.metadata,
+      spec: Lazy.any({ produce: () => this._toKube() }),
+    });
+
+    if (!this.isValidTimezone(props.timeZone)) {
+      throw new Error(`Invalid timezone: ${props.timeZone}`);
+    }
+
+    this.jobProperties = props.jobProperties;
+    this.schedule = props.schedule;
+    this.timeZone = props.timeZone;
+    this.concurrencyPolicy = props.concurrencyPolicy;
+    this.startingDeadline = props.startingDeadline;
+    this.suspend = props.suspend;
+    this.successfulJobsRetained = props.successfulJobsRetained;
+    this.failedJobsRetained = props.failedJobsRetained;
+  }
+
+  /**
+   * @internal
+   */
+  public _toKube(): k8s.CronJobSpec {
+    return {
+      concurrencyPolicy: this.concurrencyPolicy ?? ConcurrencyPolicy.FORBID,
+      failedJobsHistoryLimit: this.failedJobsRetained,
+      jobTemplate: {
+        metadata: this.jobProperties.metadata,
```

Comment:
Makes sense, will update it.

Reply:
Hey Eli, I see we can accept metadata in JobProperties right now. Is that what you are referring to or I need to introduce `jobMetadata` within `jobProperties`?

Reply:
So `metadata` and `podMetadata` will be automatically available by extending `JobProperties`. You only need to add `jobMetadata`.

Reply:
Hmm, I am still not sure if I understand. Will sync with you tomorrow about this.

From my understanding, `this.jobProperties.metadata` is the `jobMetadata` and this can right now be passed in JobProperties.

Reply:
And looks like the `k8s.CronJobSpec` does not have a metadata field like the `JobSpec` 🤔